File: Day1/calibration.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int calcCalibration(char string[150]) {
    char res1 = '0';
    char res2 = '0';

    for (int i = 0; i < strlen(string); i++) {
        if (isdigit(string[i])) {
            res1 = string[i];
            break;
        }
    }

    for (int i = strlen(string) - 1; i >= 0; i--) {
        if (isdigit(string[i])) {
            res2 = string[i];
            break;
        }
    }

    return atoi(strcat(&res1, &res2));
}

int main() {
    FILE *file;
    file = fopen("Day1/input.txt", "r");
    char line[150];
    if (file == NULL) {
        perror("Unable to open file.\n");
        return 1;
    }

    int caliber, i = 1, calibers = 0;

    while (fgets(line, sizeof(line), file) != NULL) {
        caliber = calcCalibration(line);
        printf("the caliber of line %d is %d\n", i, caliber);
        calibers += caliber;
        i++;
    }

    printf("the sum of the calibrations is: %d\n", calibers);

    fclose(file);

    return 0;
}

File: Day1/calibration.py
def calcCalibration(string):
    for char in string:
        if char.isdigit():
            i = char
            break
    
    for char in string[::-1]:
        if char.isdigit():
            i += char
            break
    
    return i




def tryCalcCalibration():
    i = 0
    sum = 0
    with open('Day1/input.txt') as file:
        for line in file:
            caliber = calcCalibration(line)
            i = i+1
            print(f"the calibration of line {i} is : {caliber}")
            sum += int(caliber)
                
        print(f"the sum of the calibers is : {sum}")
    return

tryCalcCalibration()